synapse/handlers/user_directory.py

```diff
@@ -341,11 +341,13 @@ async def _handle_new_user(
             room_id: The room ID that user joined or started being public
             user_id
         """
-        logger.debug("Adding new user to dir, %r", user_id)
-
-        await self.store.update_profile_in_user_dir(
-            user_id, profile.display_name, profile.avatar_url
-        )
+        # Local users' directory entries are created and updated in tandem with
+        # their global profile, not in response to room events.
+        if not self.is_mine_id(user_id):
+            logger.debug("Adding new user to dir, %r", user_id)
+            await self.store.update_profile_in_user_dir(
+                user_id, profile.display_name, profile.avatar_url
+            )
```

Comment:
that seems reasonable too, suspect you'll rip it out when we do remote users properly but LGTM for now